### ISSUE TITLE
🚀 Feature: Add Token to Team Membership Create event.

### ISSUE THREAD

**Issue by TheMardy** [appwrite/appwrite#3115]
### 🔖 Feature description

Like `account.recovery.create` and `account.verification.create`, send the secret / token that is required to accept a team membership when the `teams.memberships.create` event is triggered.
Possibly append it to the membership object that is returned at the moment? 
(With the same security measures as https://appwrite.io/docs/models/token)

### 🎤 Pitch

I found https://github.com/appwrite/appwrite/issues/730, which adds the token response to the password recovery and email verification endpoint. It would be great to be able to get the secret / token for a team invite as well. Adding this feature would enable Appwrite users to use their own Function for mailing.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by Meldiron** [comment]
Same comment as here: https://github.com/appwrite/appwrite/issues/3120#issuecomment-1103992105

I like it, but I am not sure how a developer would disable the original Appwrite functionality. If you want to use the custom channel to share it (for example custom email, SMS, or Discord bot), you need to disable the original email so you can ensure they got a secret from your channel.

**Comment by TheMardy** [comment]
I was thinking of the same solution: set any string as SMTP to avoid the error.
The addition of a way to disable SMTP would be a nice-to-have, but we can get custom channels to work without it. As long as we can retrieve the secret.

**Comment by gepd** [comment]
As I commented in my issue, use wrong data is not a good solution as you will keep needing the smtp to recover the super admin account, however the behavior we need is already in appwrite; We should have the same behavior as when we use the server sdk. Even with the right smtp data, the admin sdk doesn't send emails to the user when client sdk does.

I haven't seen the code but if this is some kind of flag detecting the sdk, it could have and extra var setting this value (to use system or dev emails)

What you think about it?

**Comment by TheMardy** [comment]
I agree. It's unpractical to set any string because it would not allow super admin recovery. 
I think the best way would maybe be to store this flag per project, add a toggle to "Services" in Project Settings (and maybe even store SMTP credentials per project).

**Comment by stnguyen90** [comment]
One reason why the secret isn't in the event is it's not in the [Membership Model](https://github.com/appwrite/appwrite/blob/661581a7f362f7aa26ba8e243b8542a3f2c9a9b4/src/Appwrite/Utopia/Response/Model/Membership.php#L10). If we add it to the model, though, it may end up in the API response...

**Comment by eldadfux** [comment]
Stale issue, closing down.
